### ISSUE TITLE
Offline Protocol for Android

### ISSUE THREAD

**Issue by washere** [shmolf/obsidian-editor-background#12]
Thanks for this, it's very cool.

It's not clear if this supports offline image linking. This works well on Android for online images.

Can you give us as example of how to link an offline image on Android please?

I tried these and they didn't work:

/storage/emulated/0/Download/black.png

file://storage/emulated/0/Download/black.png

It would help with some nice themes that only need a change of background color, while being offline.

I noticed it caches last image while offline. However a local storage text_field in options plus example for Android would be fantastic.

Many nice themes for Markdown with many nice color elements & great fonts are ruined simply because their background is not black or dark enough.

Thanks.

**Comment by shmolf** [comment]
I could not find a good solution that supported a local path.

However, [CSS does support base64](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs).

Here's a URL of what I used to acquire the base64. _Note_: if the image was too big, Android failed to copy the text to my clipboard.


Example of the start of a base64, with the image it represents in the background.
![Screenshot_20230820_200339_Obsidian.jpg](https://github.com/shmolf/obsidian-editor-background/assets/6433103/f989fd7d-4448-44d9-a344-35e04b6c6646)

I would much rather be able to use a local file path, rather than base64, so if you ever find a solution, please share.

**Comment by washere** [comment]
Thanks Buddy.
That's useful to know.
I used this tiny black png:
https://upload.wikimedia.org/wikipedia/commons/7/71/Black.png

It seems to cache it for offline.

I'm testing different themes.
Preferably those with black background for both editor & preview.
+
Those that have settings in:
Style Settings (community plug-in)

Most themes don't have options in Style Settings.
I like Sandstorm theme, black too, but font is white. I prefer peachy Patel pink or yellow on black.

The theme Shimmering Focus has lots of settings in Style Settings. 
The theme ion has less, but I use ion because of soft peachy pink font color for main text. It has glowing effect for heading titles etc which is a pain, but in style Settings it can be turned to zero, it's called Bloom.
I use markdown sample files which have all or most elements of Markdown to see how each theme looks.

Most themes actually don't have color options for most markdown elements. Shimmering Focus & ion are pretty good & designed colors for most.

It's stupifying how color for main font is almost impossible to assign unless making a theme:
https://forum.obsidian.md/t/change-color-of-a-word-text/14491/8

Is there an easy way to modify a theme to change the main (body text) font color?

This guy says it's way. But seems for desktop (I mainly use obsidian & everything else on Linux distros, not using Windows really), not Android, which is what I'm playing on tonight.

I think if someone made a plug-in for changing main (body text) font color, it would be very popular. But I think it's not possible for Android, even if possible for desktop versions. It would be cool for desktop-only versions too. As many plugins only work on desktop versions, when trying to install on Android, give such message.

Anyway I tested all themes or almost all, over 95% of them.
So I settled on Sandstorm, Shimmering Focus & ultimately on ion. 
Obsidian long loading time actually makes me use other Markdown apps.
If I find a way around local file linkage, I'll comment here.

But I'm closing this issue now & thanks very much for the reply. 👍
